### PR TITLE
Add method fetch for Hyperclient::Resource

### PR DESCRIPTION
**Pull request by alabeduarte**
Hi all,

Sometimes I use the [Hyperclient::Resource#[]](https://github.com/codegram/hyperclient/blob/master/lib/hyperclient/resource.rb#L56) as a hash, but I would like to avoid if statements when the attribute does not exist for some reason.

I hope this idea fit to the gem purposes

Thanks!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ### 0.7.1 (Next)
 
 * [#87](https://github.com/codegram/hyperclient/pull/87): Fix: eager delegation causes link skipping - [@dblock](https://github.com/dblock).
+* [#89](https://github.com/codegram/hyperclient/issues/89): Add method fetch for Hyperclient:: Resource - [@alabeduarte](https://github.com/alabeduarte)
```
Comment (dblock) [inline CHANGELOG.md]: Nitpicks ;)

There's a space after `::`.
Missing a period at the end.

fix via `--amend` please.

**Comment by dblock** [comment]
I like it. It should probably be documented in README as well.

**Comment by alabeduarte** [comment]
Thanks @dblock 

Just did it. I've done documented in README as you suggested.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ### 0.7.1 (Next)
 
 * [#87](https://github.com/codegram/hyperclient/pull/87): Fix: eager delegation causes link skipping - [@dblock](https://github.com/dblock).
+* [#89](https://github.com/codegram/hyperclient/issues/89): Add method fetch for Hyperclient::Resource - [@alabeduarte](https://github.com/alabeduarte)
```
Comment (dblock) [inline CHANGELOG.md]: My period is still missing! Making me nervous :)

File: README.md [hunk 3]
```diff
@@ -95,6 +95,15 @@ api.splines.each do |spline|
 end
 ```
 
+Other methods, including `[]` and `fetch` are also available
+
+```ruby
+api.users.each do |user|
```
Comment (dblock) [inline README.md]: Can we make this example based on splines like the one above? Just a more consistent story.

File: CHANGELOG.md (hunk 2 above)
Comment (alabeduarte) [inline CHANGELOG.md]: I'm sorry but I didn't understand what you meant. Can you elaborate more?

Comment (alabeduarte) [inline CHANGELOG.md]: I added the missing period, it was what you meant? :)

File: README.md (hunk 3 above)
Comment (alabeduarte) [inline README.md]: I used `uui` and `reticulated` attributes, sounds good?

**Comment by dblock** [comment]
Great, merging.

**Comment by dblock** [comment]
Btw, since it has been a while since we cut a release, I shipped 0.7.1 with this feature.

**Pull request merged**